src/csharp/Microsoft.Spark.UnitTest/UdfSerDeTests.cs

```diff
@@ -0,0 +1,178 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization.Formatters.Binary;
+using Microsoft.Spark.Utils;
+using Xunit;
+
+namespace Microsoft.Spark.UnitTest
+{
+    public class UdfSerDeTests
+    {
+        [Serializable]
+        private class TestClass
+        {
+            private readonly int _i;
+
+            public TestClass(int i)
+            {
+                _i = i;
+            }
+
+            public int MultiplyBy(int i)
+            {
+                return _i * i;
+            }
+
+            public override bool Equals(object obj)
+            {
+                var that = obj as TestClass;
+
+                if (that == null)
+                {
+                    return false;
+                }
+
+                return this._i == that._i;
+            }
+
+            public override int GetHashCode()
+            {
+                return base.GetHashCode();
+            }
+        }
+
+        [Fact]
+        public void TestUdfSerDe()
+        {
+            {
+                // Without Closure
+                Func<int, int> udf = i => 10 * i;
+                VerifyUdfSerDe(udf, false, false);
+            }
+
+            {
+                // With Closure
+                TestClass tc = new TestClass(20);
+                Func<int, int> udf = i => tc.MultiplyBy(i);
+                VerifyUdfSerDe(udf, true, false);
```

Comment:
The lambda will create an anonymous class as the target, where the field of the target will be TestClass.  I set it to false because there didn't seem to be an easy way of defining a custom equals method for lambda functions.   I modified the test to instead just check the target fields for equality.